quickfixj-core/src/main/java/quickfix/ThreadedSocketInitiator.java

```diff
@@ -82,11 +82,12 @@ public void stop() {
     }
 
     public void stop(boolean forceDisconnect) {
-        stopInitiators();
         logoutAllSessions(forceDisconnect);
+        stopSessionTimer();
         if (!forceDisconnect) {
             waitForLogout();
```

Comment:
This isn't required because `logoutAllSessions(forceDisconnect)` does exactly the same, unless the sessions weren't initialised... in which case any point waiting for logout?